### PR TITLE
Clip

### PR DESCRIPTION
**Pull request by seemantk**
Adds a clip path to the notesbook `<g>`.

### CODE REVIEW

**Comment by seemantk** [comment]
paging @curran

**Comment by curran** [comment]
Changes look great @seemantk . Will review and test today.

File: js/notesbook.js [hunk 1]
```diff
@@ -106,8 +106,18 @@ function NotesBook() {
           .call(mensurationsRender)
       ;
       g
+        .append("clipPath")
+          .attr("id", "noteclip")
+        .append("rect")
+          .attr("x", 0)
```
Comment (curran) [inline js/notesbook.js:113]: Nice work adding the clip path! It seems like it should be getting applied to the inner rectangle, inside the margin, so the labels don't overlap with the notes, but that's not what I'm seeing when I run it.

image

I wonder what's going on here..

File: js/notesnav.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ function NotesNav() {
         ;
         brush.selection = g.select(".brush")
             .call(brush.widget)
-            brush.selection.call(brush.widget.move, canvas.widget.x().range())
+            .call(brush.widget.move, canvas.widget.x().range())
```
Comment (curran) [inline js/notesnav.js:58]: Nice catch.

File: js/notesbook.js [hunk 3]
```diff
@@ -106,8 +106,18 @@ function NotesBook() {
...
+          .attr("y", 0)
+          .attr("width", width)
+          .attr("height", height)
+      ;
+      g
         .append("g")
           .attr("class", "notesbook")
+          .attr("clip-path", "#noteclip")
```
Comment (curran) [inline js/notesbook.js]: I believe this needs to be `.attr("clip-path", "url(#noteclip)")`. Based on docs at https://developer.mozilla.org/en/docs/Web/SVG/Element/clipPath

File: js/notesnav.js [hunk 4]
```diff
@@ -66,29 +66,33 @@ function NotesNav() {
...
+                .call(brush.widget.move, extent)
+            ;
+        }
+        if(dispatch)
+            dispatch.call("zoom", this, extent.map(canvas.widget.x().invert));
     } // brushed()
 
-    function extent() {
-    } // extent()
+    function recenter(clickX) {
```
Comment (curran) [inline js/notesnav.js:84]: This is nice. Interestingly, Mike Bostock recently re-did the focus + context example, and he made the brush visible all the time https://bl.ocks.org/mbostock/34f08d5e11952a80609169b7917d4172

**Review by curran — APPROVED** [review]
Great work @seemantk ! One small change is required for the clip path to work, I can add it.

File: js/notesbook.js (hunk 3 above)
Comment (seemantk) [reply js/notesbook.js]: Makes sense, I was wondering why it didn't take effect as I thought.  Do you mind to make the change, as I'm having computer issues at the moment.

**Comment by curran** [comment]
After enabling the clip path with the `url(#noteclip)` syntax, I see that there are a few issues here:
- reflines are also clipped out, as they are DOM children of the `<g>` that the clip path is applied to.
- barlines are not clipped

image

**Comment by curran** [comment]
So I moved the clip path down the DOM tree into a new `<g>` created to house the notes rects only. Here's what we have now:

image

The last remaining thing is to also clip out the bar lines.

**Comment by curran** [comment]
Clipped out the bar lines, I think this is good to go!

image

**Pull request merged**